### PR TITLE
:tada: Tiktok marketing - remove odd auth method

### LINKED ISSUE #11029: [EPIC] Source TikTok to Cloud Beta
According to [checklist](https://docs.google.com/spreadsheets/d/1L4LNVgd72nM6BjzWKpqWtiM6EVh8Cp5lkl1BKYBACAw/edit#gid=1288588270) improvements:

- MLP requirements as surfaced through [research](https://docs.google.com/document/d/1CZ-0H9BfNaP7FQrts2MnpGFkilvLwCqulhnLCrDo0mA/edit):
  - [ ] #11368

- Engineering Acceptance Criteria:
   - [x] #11372
   - [x] #11376
   - [x] #11377

- Outstanding bugs:
   - [x] https://github.com/airbytehq/airbyte/issues/9709
   - [x] https://github.com/airbytehq/airbyte/issues/9491
   - [x] https://github.com/airbytehq/airbyte/issues/6875 
   - [x] #11375
   - [x] #12107

### LINKED ISSUE #12499: Source tiktok ads: Consolidate the `Oauth2.0` and `Production Access Token` auth types
## Current Behavior
Both auth methods seem to exhibit the exact same behavior. So we should remove one of them

## Expected Behavior
remove the `Production Access Token` auth method and update the docs accordingly

### PR DESCRIPTION
**Pull request by midavadim**
## What
*Describe what the change is solving*
*It helps to add screenshots if it affects the frontend.*

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Review by annalvova05 — APPROVED** [review]

File: airbyte-integrations/connectors/source-tiktok-marketing/integration_tests/spec.json [hunk 1]
```diff
@@ -5,43 +5,13 @@
     "title": "TikTok Marketing Source Spec",
     "type": "object",
     "properties": {
-      "credentials": {
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-tiktok-marketing/integration_tests/spec.json:8]: this will break existing configs, so we shouldn't rename it

**Review by sherifnada — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-tiktok-marketing/integration_tests/spec.json (hunk 1 above)
Comment (midavadim) [reply airbyte-integrations/connectors/source-tiktok-marketing/integration_tests/spec.json:8]: All OLD auth_types:
    prod_access_token
    sandbox_access_token
    oauth2.0

#1. If I just remove the odd auth method (oauth2.0) then old "OAuth2.0" related config fails validation.

So in order to support all possible old configs I renamed 'credentials' to 'credentials_all'.
It allows successful validation for all possible configs.


#2. But if we know that all old "OAuth2.0" config can be updated to new format (and use prod_access_token instead) then 'credentials' can be left unchanged.

@sherifnada   what do you think about #2, I don't know how we manage "OAuth2.0" in airbyte cloud.

Comment (sherifnada) [reply airbyte-integrations/connectors/source-tiktok-marketing/integration_tests/spec.json:8]: @midavadim my apologies I didn't see the application logic change below. I see now how this was maintaining backwards compatibility. 

IMO the simpler thing to do here would be to remove the `prod_access_token` -- my guess is most people are using oauth or sandbox. if it fails, they'll have to fix it, but renaming the whole object means that everyone using tiktok will have to re-set their configuration in the UI. I think until we have migrations that's a better outcome than having an old branch in the code that lives forever. So I would support removing prod_access_token

File: airbyte-oauth/src/test/java/io/airbyte/oauth/flows/BaseOAuthFlowTest.java [hunk 2]
```diff
@@ -197,6 +197,18 @@ protected String getConstantState() {
...
 
+  protected String getMockedResponse() {
+    final Map<String, String> returnedCredentials = getExpectedOutput();
+    return Jsons.serialize(returnedCredentials);
+  }
+
+  protected OAuthConfigSpecification getOAuthConfigSpecification() {
+    return getoAuthConfigSpecification()
+        // change property types to induce json validation errors.
+        .withCompleteOauthServerOutputSpecification(getJsonSchema(Map.of("client_id", Map.of("type", "integer"))))
```
Comment (midavadim) [inline airbyte-oauth/src/test/java/io/airbyte/oauth/flows/BaseOAuthFlowTest.java:208]: @etsybaev  I was sure that client_id and secret should be a 'string'? Could you please re-check

File: airbyte-integrations/connectors/source-tiktok-marketing/integration_tests/spec.json [hunk 3]
```diff
@@ -41,36 +41,7 @@
             },
             "required": ["app_id", "secret", "access_token"]
           }, {
-            "title": "Production Access Token",
```
Comment (midavadim) [inline airbyte-integrations/connectors/source-tiktok-marketing/integration_tests/spec.json:44]: @sherifnada 

'Production' - is completely removed. 

Note: In order to pass all tests I have to update 5 config files (replace "prod_access_token" to "oauth2.0")

File: airbyte-oauth/src/test/java/io/airbyte/oauth/flows/TikTokMarketingOAuthFlowTest.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
+/*
```
Comment (midavadim) [inline airbyte-oauth/src/test/java/io/airbyte/oauth/flows/TikTokMarketingOAuthFlowTest.java:1]: this is just a completely new java test. (it was missing before)

Comment (sherifnada) [reply airbyte-oauth/src/test/java/io/airbyte/oauth/flows/TikTokMarketingOAuthFlowTest.java:1]: good addition!

**Review by sherifnada — APPROVED** [review]
spec change lgtm, make sure to pull master since it had some updates to tiktok (if you haven't already)

**Comment by sherifnada** [comment]
@midavadim shouldn't this be merged? :)

**Comment by midavadim** [comment]
> 

@sherifnada it was merged 10 mins before your comment :)

**Comment by sherifnada** [comment]
Screen Shot 2022-05-06 at 3 25 57 PM
ah sorry I am seeing it as "closed" not merged that's all i meant

**Pull request merged**